### ISSUE TITLE
[Ch 16][Notebook][Major] Loss 노트 "숫자로 감 잡기" 표 — K=7 softmax·손실 값이 계산과 다름 (4행 중 3행)

### ISSUE THREAD

**Issue by sy2399** [yoon-gu/neuqes-101#154]
## 대상

- 기준 커밋: `61a71df` (origin/master)
- 파일: `16_ko_multiclass/16_ko_multiclass.ipynb` c3 (📐 Loss 노트 "숫자로 감 잡기" 표), `pages/16-ko_multiclass.md:99-101`, `book/chapters/ch16_ko_multiclass.tex:138-140`
- 환경: 정적 검토 + 손계산 검산. **Colab 재실행 미수행**

## 실제 결과

"숫자로 감 잡기 (K=7, 정답 = 클래스 5)" 표의 **4행 중 3행이 계산과 맞지 않습니다.** 첫 행(균등)만 맞습니다.

| logits | 표의 $\hat p_5$ | 표의 손실 | **실제** $\hat p_5$ | **실제** 손실 |
|---|---|---|---|---|
| 모두 0 | 0.143 | 1.946 | 0.143 | 1.946 ✅ |
| 정답 클래스만 +2 | 0.471 | 0.752 | **0.552** | **0.594** |
| 정답 클래스만 +5 | 0.985 | 0.015 | **0.961** | **0.040** |
| 다른 클래스 +5 (정답 0) | 0.005 | 5.302 | **0.006** (하나만 +5) | **5.040** |

$\hat p = e^{z_y} / \sum_j e^{z_j}$ 에 K=7 을 넣으면 위와 같습니다. 예: $e^2/(e^2+6) = 7.389/13.389 = 0.552$, $-\ln 0.552 = 0.594$.

표의 값은 어떤 K 에서도 나오지 않습니다 (0.471 이 되려면 나머지 클래스가 8.3개여야 함). 4행은 "다른 클래스 +5" 가 *하나만* +5 인지 *여섯 전부* +5 인지도 불분명합니다 — 전부 +5 면 $\hat p = 1/(1+6e^5) = 0.001$, 손실 6.79 입니다.

## 기대 결과

```
| 정답 클래스만 +2 | $\approx 0.552$ | 0.594 |
| 정답 클래스만 +5 | $\approx 0.961$ | 0.040 |
| 다른 클래스 하나만 +5 (정답 0) | $\approx 0.006$ | **5.040** ← 자신 있게 틀림 |
```

4행 라벨은 "하나만" 을 명시합니다.

## 재현 절차

```python
import math
def loss(z, ci): s = sum(math.exp(v) for v in z); p = math.exp(z[ci]) / s; return p, -math.log(p)
loss([2,0,0,0,0,0,0], 0)   # (0.5519, 0.594)   표: 0.471 / 0.752
loss([5,0,0,0,0,0,0], 0)   # (0.9611, 0.040)   표: 0.985 / 0.015
loss([0,5,0,0,0,0,0], 0)   # (0.0065, 5.040)   표: 0.005 / 5.302
```

## 추가 확인

**Ch 12 의 같은 표(K=5)도 세 행이 틀립니다** — `(0,0,0,0,2)` 실제 0.649/0.433 (표 0.541/0.615), `(0,0,0,0,5)` 0.974/0.027 (표 0.985/0.015), `(5,0,0,0,0)` 0.007/5.027 (표 0.005/5.310). Ch 16 표가 Ch 12 표를 옮겨 오며 K 만 바꾼 흔적입니다. Ch 12 건은 별도 이슈로 올립니다.

이 표는 학습자가 **손으로 검산해 보라고 둔 교육용 예시** 입니다. 검산하면 안 맞는 표는 CE 의 직관("정답 logit 이 +2 만 돼도 손실이 1.9 → 0.6 으로 떨어진다") 자체를 흐립니다.

## 전파 범위

```bash
grep -n '0\.471\|0\.752\|5\.302' 16_ko_multiclass/16_ko_multiclass.ipynb pages/16-*.md book/chapters/ch16_ko_multiclass.tex
```

| 축 | 상태 |
|---|---|
| 📓 Notebook | c3 **3행** |
| 📄 WikiDocs | `pages/16-ko_multiclass.md:99-101` **3행** |
| 📕 EPUB | `ch16_ko_multiclass.tex:138-140` **3행** — tex 손수정 |

## 심각도

**Major** — 계획서 §9 "잘못된 수식" 에 해당합니다. 학습자가 검산하면 틀린 값을 배우거나 자기 계산을 의심하게 됩니다.

## 담당

**재실행 불필요(마크다운) · 재변환 불필요(손수정) · tex 손수정 후 EPUB 재빌드 시 반영.**

Related to #40

**Comment by sy2399** [comment]
## 수정 완료 — @sy2399 (슬롯 b2), 2026-08-28

브랜치 `fix/ch16-review-2026-09-b2` · 커밋 `b283777`

| logits | $\hat p_5$ | 손실 |
|---|---|---|
| 정답 클래스만 +2 | 0.471 → **0.552** | 0.752 → **0.594** |
| 정답 클래스만 +5 | 0.985 → **0.961** | 0.015 → **0.040** |
| 다른 클래스 **하나만** +5 (정답 0) | 0.005 → **0.006** | 5.302 → **5.040** |

4행 라벨에 "하나만" 을 넣어 여섯 전부 +5 인 경우(손실 6.79)와 구분했습니다.

| 축 | 파일 | |
|---|---|---|
| 📓 Notebook | c3 | ✅ |
| 📄 WikiDocs | `pages/16-ko_multiclass.md:99-101` | ✅ |
| 📕 EPUB | `tex:138-140` 손수정 | ✅ (재빌드 시 반영) |

Ch 12 의 같은 표(K=5)는 #161 로 따로 올렸습니다 — 그쪽은 PR #136 브랜치에서 고쳐야 합니다.

**등록자 = 수정자** 라 교차 확인이 필요합니다 — 페어의 다른 담당자가 확인 후 닫아 주세요.

**Comment by sy2399** [comment]
PR #236 (`2b2a18c`) 병합으로 master 에 반영됐습니다.

1·2차 수정 + 시드 고정 재실행 + WikiDocs 재변환 + tex·EPUB 재생성이 모두 포함된 병합이며, 3축 PASS 기록은 #40 에 있습니다.
